Source/WebCore/inspector/NetworkResourcesData.cpp

```diff
@@ -316,19 +325,24 @@ Vector<String> NetworkResourcesData::removeCachedResource(CachedResource* cached
 
 void NetworkResourcesData::clear(std::optional<String> preservedLoaderId)
 {
-    m_requestIdsDeque.clear();
-    m_contentSize = 0;
-
-    if (!preservedLoaderId)
+    if (!preservedLoaderId) {
         m_requestIdToResourceDataMap.clear();
-    else {
-        Vector<String> keysToRemove;
-        for (auto& [key, value] : m_requestIdToResourceDataMap) {
-            if (value->loaderId() != *preservedLoaderId)
-                keysToRemove.append(key);
+        m_requestIdsDeque.clear();
+        m_contentSize = 0;
+    } else {
+        ListHashSet<String> requestIdsDeque(WTFMove(m_requestIdsDeque));
+
+        for (const auto& requestId : requestIdsDeque) {
+            auto resourceData = resourceDataForRequestId(requestId);
+            if (!resourceData)
+                continue;
+            if (resourceData->loaderId() == *preservedLoaderId)
+                m_requestIdsDeque.add(requestId);
+            else {
+                m_contentSize -= resourceData->evictContent();
```

Comment:
rather than subtracting from `m_contentSize` i think it might be more accurate to recalculate `m_contentSize` from what's in `m_requestIdsDeque` after iterating (or perhaps when the `requestId` is added back to `m_requestIdsDeque`)